azurelinuxagent/ga/exthandlers.py

```diff
@@ -288,6 +298,18 @@ def handle_ext_handler(self, ext_handler, etag):
         ext_handler_i = ExtHandlerInstance(ext_handler, self.protocol)
 
         try:
+            state = ext_handler.properties.state
+            # Valid states are enabled, disabled and uninstall
+            if state == u"enabled":
+                # there is an upgrade GUID and it is NOT new
+                if ext_handler.properties.upgradeGuid is not None and \
+                        not self.is_new_guid(ext_handler):
+                    logger.info("New GUID is the same as the old GUID. Exiting without upgrading.")
+                    return
+            elif state == u"disabled" or state == u"uninstall":
+                # Remove the GUID from the dictionary in this case so that
+                # it is upgraded upon re-enabling/re-install
+                self.last_guids.pop(ext_handler.name, None)
```

Comment:
@uiri The `disable` or `uninstall` could fail. Perhaps the upgradeGuid should be removed after they've succeeded?

Reply:
If the `disable` or `uninstall` fails, should a subsequent `enable` proceed even without a new upgrade GUID? My initial thinking was "yes" but perhaps enable after a botched uninstall (or botched disable) is not a good idea.

Reply:
@uiri It seems consistent to remove the GUID only when disable / uninstall succeeds.

Reply:
OK, that makes sense. I made the change in the most recent commit.